helm/charts/oathkeeper/templates/hpa-controller.yaml

```diff
@@ -20,17 +21,34 @@ spec:
   maxReplicas: {{ .Values.deployment.autoscaling.maxReplicas }}
   metrics:
   {{- with .Values.deployment.autoscaling.targetMemory }}
+  {{- if and (and (not (empty $kubeVersion)) (semverCompare ">=1.30-0" $kubeVersion) (not (empty $.Values.deployment.autoscaling.extraMetrics))) }}
```

Comment:
to avoid breaking change for people already using extraContainers, we switch to containerResource only when it's available in kubernetes > 1.30 and Values.deployment.autoscaling.extraMetrics is defined.